trinity/sync/beam/chain.py

```diff
@@ -378,6 +414,117 @@ async def run(self) -> None:
         await self.manager.run_service(self._body_syncer)
 
 
+class BodyChainGapSyncer(Service):
+    """
+    A service to sync historical blocks without executing them. This service is meant to be run
+    in tandem with other operations that sync the state.
+    """
+
+    _starting_tip: BlockHeader = None
+    logger = get_logger('trinity.sync.beam.chain.BodyChainGapSyncer')
+
+    def __init__(self,
+                 chain: AsyncChainAPI,
+                 db: BaseAsyncChainDB,
+                 peer_pool: ETHPeerPool) -> None:
+        self._chain = chain
+        self._db = db
+        self._peer_pool = peer_pool
+        self._paused = False
+        self._resumed = asyncio.Event()
+
+    async def _setup_for_next_gap(self) -> None:
+        gap_start, gap_end = self._get_next_gap()
+        start_num = BlockNumber(gap_start - 1)
+        _starting_tip = await self._db.coro_get_canonical_block_header_by_number(start_num)
+
+        async def _get_launch_header() -> BlockHeaderAPI:
+            return _starting_tip
+
+        self.logger.debug("Starting to sync missing blocks from #%s to #%s", gap_start, gap_end)
+
+        self._body_syncer = FastChainBodySyncer(
+            self._chain,
+            self._db,
+            self._peer_pool,
+            DatabaseBlockRangeHeaderSyncer(self._db, (gap_start, gap_end,)),
+            launch_header_fn=_get_launch_header,
+            should_skip_header_fn=body_for_header_exists(self._db, self._chain)
+        )
+
+    def _get_next_gap(self) -> BlockRange:
+        gaps, future_tip_block = self._db.get_chain_gaps()
+        if len(gaps) == 0:
+            # We do not have gaps in the chain of blocks but we may still have a gap from the last
+            # block up until the highest consecutive written header.
+            header_gaps, future_tip_header = self._db.get_header_chain_gaps()
+            if len(header_gaps) > 0:
+                # The header chain has gaps, find out the lowest missing header
+                highest_consecutive_missing_header, _ = header_gaps[0]
```

Comment:
Took me a minute to parse this. Maybe:
```suggestion
                lowest_missing_header, _ = header_gaps[0]
```

Reply:
Ah yes :see_no_evil: